Enable certificate handling for MQTT TLS connections.

marcfir opened the pull request:
Enable certificate handling for MQTT TLS connections. Fully resolves issue #311. Partially resolves issues #309 and #310.

- Add mqtt configuration options for CA certificate path
  - Default for mqtt.CaTrustStorePath="./certs".
  - Default for mqtt.CaCertPath="/etc/ssl/certs/" for Linux and "./certs" for Windows
- Add tests against dev.umati.app for testing secure connections
- Add requirements to the ca-certificates Debian package
- Add CA certificates to the Windows artifacts from https://curl.se/docs/caextract.html

GoetzGoerisch reviewed (commented):
Is it possible to bring up a MQTT broker with WSS support inside the tests?
open62541 are doing this if I'm not mistaken.

Otherwise these tests rely on external resources.

marcfir commented:
> Is it possible to bring up a MQTT broker with WSS support inside the tests? open62541 are doing this if I'm not mistaken.
> 
> Otherwise these tests rely on external resources.

Yes, an integration test based on Docker Compose against an Eclipse Mosquitto broker is possible. It uses self-signed certificates. I added the test and removed the old unit test.

GoetzGoerisch reviewed (approved):
Great! Thank you!
